smt_parser.py

```diff
@@ -1,3 +1,4 @@
+from __future__ import annotations
```

Comment:
i haven't reviewed the parsing. So long as you don't parse a `sat` as an `unsat`, you can't break soundness.